### PR TITLE
gh-87092: move assembler related code from compile.c to assemble.c

### PR DESCRIPTION
**Pull request by iritkatriel**
* Issue: gh-87092

### CODE REVIEW

File: Include/internal/pycore_flowgraph.h [hunk 1]
```diff
@@ -91,10 +92,9 @@ void _PyCfgBuilder_Fini(_PyCfgBuilder *g);
 
 _PyCfgInstruction* _PyCfg_BasicblockLastInstr(const _PyCfgBasicblock *b);
 int _PyCfg_OptimizeCodeUnit(_PyCfgBuilder *g, PyObject *consts, PyObject *const_cache,
-                            int code_flags, int nlocals, int nparams);
+                            int code_flags, int nlocals, int nparamsi, int firstlineno);
```
Comment (iritkatriel) [inline Include/internal/pycore_flowgraph.h]:
```suggestion
                            int code_flags, int nlocals, int nparams, int firstlineno);
```

File: Python/compile.c [hunk 2]
```diff
@@ -7389,13 +6802,17 @@ add_return_at_end(struct compiler *c, int addNone)
     return SUCCESS;
 }
 
+static int cfg_to_instr_sequence(cfg_builder *g, instr_sequence *seq);
 
 static PyCodeObject *
 assemble_code_unit(struct compiler_unit *u, PyObject *const_cache,
```
Comment (markshannon) [inline Python/compile.c]: Is this name still correct? Was it ever?

It seems to do optimization and assembly.

**Review by markshannon — COMMENTED** [review]
Seems reasonable from a fairly quick scan.

There is no change in capability or optimizations, this is just a refactor. Correct?

File: Python/compile.c (hunk 2 above)
Comment (iritkatriel) [reply Python/compile.c]: There used to be just assemble(C) which takes the compiler and and does all the optimization and assembly. I split out assemble_code_unit (which takes just the c->u and not the whole c) and does most of what used to be in assemble().   So the name was always incorrect (assemble() called the optimizer).   I think the reason it was done like this is that assemble() gets called from a number of places in the compiler (lambda, class, module, etc).   So it was convenient to have one function that does opt+assemble and returns a code object. 

We can rename assemble() and assemble_code_unit(). Any suggestions what to call it? something like instr_sequence_to_code_object()?

**Review by markshannon — APPROVED** [review]

**Pull request merged**